src/priority-queue-original.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * A priority queue stores a list of items but each can have a numeric priority value.
+ * Items with a higher priority are dequeued before items with a lower priority.
+ * Implemented as a hash of arrays where the hash keys are priority values.
+ */
+function PriorityQueue(size) {
+	this.store = {};	// keys are priorities, values are arrays of elements
+	this.count = 0;
+
+	// adds an item
+	// priority must be an integer (higher value has higher priority)
+	this.add = function(value, priority) {
+		if (this.store[priority] == undefined)
+			this.store[priority] = [];
+
+		this.store[priority].push(value);
+		this.count++;
+	};
+
+	// returns the oldest-added value with the highest priority
+	this.Pop = function() {
+		maxKey = Math.max(Object.keys(this.store));
+		this.count--;
+		return this.store[maxKey].shift();
+	};
+
+	this.length = function() {
+		return this.count;
+	}
+}
+
+PriorityQueue.prototype.get_all_priorities = function() {
```

Comment:
```suggestion
PriorityQueue.prototype.getPriorities = function() {
```
Convert this from snake_case (which gets its name from Python 🐍) to superior (and more importantly, standard) camelCase 🐫. I would also remove the word `All` from this unless it's meaningful.

If this was part of a class, we could also make this a getter.